dcrlibwallet.go

```diff
@@ -100,6 +100,8 @@ func newLibWallet(walletDataDir, walletDbDriver string, activeNet *netparams.Par
 		syncData:      syncData,
 	}
 
+	listenForShutdown()
```

Comment:
Let's tie the shutdown listener to the `lw` instance to ensure proper deallocation of variables when the `lw` instance dies. This also means adding the `shuttingDown` channel and `cancelFuncs` slice to the `LibWallet` struct.